config/initializers/warden.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+Rails.configuration.middleware.use RailsWarden::Manager do |manager|
+  manager.failure_app = ->(_env) {
+    [401, { 'Content-Type' => 'application/json' }, [{ message: 'Not authorized', code: 401 }.to_json]]
+  }
+
+  manager.default_strategies :http_header_auth
+end
+
+Warden::Strategies.add(:http_header_auth) do
+  def valid?
+    request.env[Rails.application.config_for(:warden)['remote_user_header']].present?
+  end
+
+  def authenticate!
+    user = request.env[Rails.application.config_for(:warden)['remote_user_header']]
+    authorized_users = Rails.application.config_for(:warden)['authorized_users'].split(',')
+    if authorized_users.include?(user)
+      email = env['HTTP_X_AUTH_REQUEST_EMAIL']
+      success! GUIUser.find_or_create_by!(email:)
```

Comment:
It looks like you're missing the actual argument here.

Reply:
I believe that if the key & argument are named the same, you don't have to type out the argument & rubocop will ding you for it if you do. When I tested it out it seems to work either way, but I'm fine with adding it if that makes more sense to read.

Reply:
Oh, oops - sorry, my mistake! I'm behind the times.